### PR TITLE
[Epg] Add missing Epg tags

### PR DESCRIPTION
**Pull request by Angelinas1**
- Add missing Epg tags 
 (OriginalTitle,Cast,Director,Writer,Year,IMDBNumber,EpisodeId,EpisodePart,EpisodeName)

### CODE REVIEW

**Comment by phunkyfish** [comment]
This conflicts with https://github.com/kodi-pvr/pvr.iptvsimple/pull/243

Both PRs use different fields for episode name. Which one is correct?

File: src/PVRIptvData.cpp [hunk 1]
```diff
@@ -274,11 +274,24 @@ bool PVRIptvData::LoadEPG(time_t iStart, time_t iEnd)
...
+    entry.strWriter = "";
+    entry.iYear = 0;
+    entry.strIMDBNumber = "";
+    entry.strEpisodeName = "";
     entry.startTime = iTmpStart;
     entry.endTime = iTmpEnd;
 
     GetNodeValue(pChannelNode, "title", entry.strTitle);
     GetNodeValue(pChannelNode, "desc", entry.strPlot);
+    GetNodeValue(pChannelNode, "sub-title", entry.strOriginalTitle);
```
Comment (ksooo) [inline src/PVRIptvData.cpp:300]: This should read `entry.episodeName`.

File: src/PVRIptvData.cpp [hunk 2]
```diff
@@ -274,11 +274,24 @@ bool PVRIptvData::LoadEPG(time_t iStart, time_t iEnd)
...
+    GetNodeValue(pChannelNode, "actor", entry.strCast);
+    GetNodeValue(pChannelNode, "director", entry.strDirector);
+    GetNodeValue(pChannelNode, "writer", entry.strWriter);
+    GetNodeValue(pChannelNode, "Id", entry.strIMDBNumber);
+    GetNodeValue(pChannelNode, "episode-num", entry.strEpisodeName);
```
Comment (ksooo) [inline src/PVRIptvData.cpp:309]: Not sure what `episode-num` is defined to contain, but the name suggests this is more the number of the episode and not its name?

So, maybe mapping `episode-num` to `entry.iEpisodeNumber ` would be correct, no?

File: src/PVRIptvData.h [hunk 3]
```diff
@@ -35,10 +35,17 @@ struct PVRIptvEpgEntry
   int         iChannelId;
   int         iGenreType;
   int         iGenreSubType;
+  int         iYear;
```
Comment (ksooo) [inline src/PVRIptvData.h:38]: You seem not to fill the `iYear` member, so no need to add this member.

**Review by ksooo — CHANGES_REQUESTED** [review]
This needs some more love.

File: src/PVRIptvData.cpp (hunk 1 above)
Comment (Angelinas1) [reply src/PVRIptvData.cpp:300]: sub-title is for non english user...see this thread
https://forum.kodi.tv/showthread.php?tid=201981
`<sub-title lang="fr">Le brasier</sub-title> `

File: src/PVRIptvData.cpp (hunk 2 above)
Comment (Angelinas1) [reply src/PVRIptvData.cpp:309]: Episodename can't be integer, in epg use letter to see what is season and episode number:
example:
`<episode-num system="onscreen">Staffel 2, Folge 32</episode-num> `

File: src/PVRIptvData.h (hunk 3 above)
Comment (Angelinas1) [reply src/PVRIptvData.h:38]: I was try but have bug...little help need for this :)

**Comment by Angelinas1** [comment]
test was ok, mine Epg12.db is good populated:

**Comment by phunkyfish** [comment]
But you don’t use sub title (episode name) or episode number in your screenshot which is what the review refers to.

**Review by Angelinas1 — COMMENTED** [review]
> 
> 
> But you don’t use sub title (episode name) or episode number which is what the review refers to.



> 
> 
> But you don’t use sub title (episode name) or episode number which is what the review refers to.

Yes, I use from code
sub-title from epg.xml  is Original name see on image #1
*King Arthur: Legend of the Sword (title)
*Kralj Arthur: Legenda o maču (OriginalName), this is same like title but on mine language

here is db image for episodename

File: src/PVRIptvData.cpp (hunk 2 above)
Comment (Sorien) [reply src/PVRIptvData.cpp:309]: according specs https://github.com/XMLTV/xmltv/blob/master/xmltv.dtd#L363 it can be

**Comment by Sorien** [comment]
it would me nice if you could implement it according official specs https://github.com/XMLTV/xmltv/blob/master/xmltv.dtd

File: src/PVRIptvData.cpp (hunk 2 above)
Comment (Angelinas1) [reply src/PVRIptvData.cpp:309]: That is episode number , in Kodi that tag is iEpisodeId).I didnt add that.
Mine is strEpisodename:)

Comment (phunkyfish) [reply src/PVRIptvData.cpp:309]: But why store episode-num from the XMLTV into strEpisodeName in kodi pvr? 

Can someone give some examples of what episode-num fields look like from some XMLTV? Does it need to be parsed first?

File: src/PVRIptvData.h (hunk 3 above)
Comment (phunkyfish) [reply src/PVRIptvData.h:38]: Do you mean you don’t understand how to populate/parse iYear?

File: src/PVRIptvData.cpp (hunk 1 above)
Comment (phunkyfish) [reply src/PVRIptvData.cpp:300]: That’s incorrect, sub-title Is for any language including English. @ksooo is right it should be stored in Episode name.

File: src/PVRIptvData.cpp (hunk 2 above)
Comment (phunkyfish) [reply src/PVRIptvData.cpp:309]: From reading the spec this needs to parsed.

Comment (Sorien) [reply src/PVRIptvData.cpp:309]: @phunkyfish i have some for czech tv stations http://phazebox.com/epg.php?user=dp&email=dp@dp.sk but donno what's the source 

and http://xmltv.xmltv.se is a good source

http://xmltv.xmltv.se/nature-history.hr.viasatworld.com_2019-06-08.xml
http://xmltv.xmltv.se/family.cinema.sky.de_2019-06-26.xml

File: src/PVRIptvData.cpp (hunk 1 above)
Comment (phunkyfish) [reply src/PVRIptvData.cpp:300]:
```
  <programme start="20190616053000 +0200" stop="20190616053000 +0200" channel="EDGEsport">
    <title lang="en">Live Recap Asia Talent Cup 2018</title>
    <sub-title lang="en">Malaysia - Race 2</sub-title>
    <desc lang="en">Live Recap. Relive the action from the Asia Talent Cup. The competition sees Asia's stars of the future participate in a ten race championship held over five rounds throughout the year, with all of the riders on homogenised Honda NSF250R Moto3™ bikes(n)</desc>
  </programme>
```

In the above sub-title is the episode name.

File: src/PVRIptvData.cpp (hunk 2 above)
Comment (phunkyfish) [reply src/PVRIptvData.cpp:309]: Ok, there seems to be two types of episode-num system and onscreen from a scan of the samples. The format fro onscreen does not appear to be consistent or at the very least can be language dependent.

```
<episode-num system="themoviedb.org">movie/17264</episode-num>
```

```
<episode-num system="onscreen">S7 E2</episode-num>
```

**Comment by Angelinas1** [comment]
screenshot005
tnx for all your time...next time I will create with proper github desktop aplication and with Love.

**Pull request closed without merging**